### PR TITLE
Support AutoIndex

### PR DESCRIPTION
**Pull request by longjiquan**
/kind improvement

### CODE REVIEW

File: internal/proxy/task_index.go [hunk 1]
```diff
@@ -104,7 +105,22 @@ func (cit *createIndexTask) OnEnqueue() error {
 	return nil
 }
 
+func wrapUserIndexParams(metricType string) []*commonpb.KeyValuePair {
+	return []*commonpb.KeyValuePair{
+		{
+			Key:   common.IndexTypeKey,
+			Value: "AutoIndex",
```
Comment (czs007) [inline internal/proxy/task_index.go:112]: AutoIndex or  AUTOIndex?

File: internal/proxy/task_index.go [hunk 2]
```diff
@@ -142,9 +158,57 @@ func (cit *createIndexTask) parseIndexParams() error {
...
+
+				return nil
+			}
+
 			if !exist {
-				return fmt.Errorf("IndexType not specified")
+				if err := handle(0); err != nil {
+					return err
+				}
+			} else if specifyIndexType == "AutoIndex" {
```
Comment (czs007) [inline internal/proxy/task_index.go:208]: change AutoIndex to const definition

File: internal/proxy/task_index.go [hunk 3]
```diff
@@ -142,9 +158,57 @@ func (cit *createIndexTask) parseIndexParams() error {
...
+					return nil
+				}
+
+				metricType, metricTypeExist := indexParamsMap[common.MetricTypeKey]
+
+				if len(indexParamsMap) > numberParams+1 {
+					return fmt.Errorf("only metric type can be passed when use AutoIndex")
+				}
+
+				if len(indexParamsMap) == numberParams+1 && !metricTypeExist {
```
Comment (czs007) [inline internal/proxy/task_index.go:189]: merge 189 ~ 200 to  one `if` statement.

File: internal/proxy/task_index.go [hunk 4]
```diff
@@ -142,9 +158,57 @@ func (cit *createIndexTask) parseIndexParams() error {
 			for k, v := range Params.AutoIndexConfig.IndexParams.GetAsJSONMap() {
 				indexParamsMap[k] = v
 			}
-		} else {
```
Comment (czs007) [inline internal/proxy/task_index.go:145]: add comment for `if` case.

File: internal/proxy/task_index.go [hunk 5]
```diff
@@ -134,17 +150,65 @@ func (cit *createIndexTask) parseIndexParams() error {
 
 	if isVecIndex {
 		specifyIndexType, exist := indexParamsMap[common.IndexTypeKey]
-		if Params.AutoIndexConfig.Enable.GetAsBool() {
+		if Params.AutoIndexConfig.Enable.GetAsBool() { // `enable` only for cloud instance.
```
Comment (chasingegg) [inline internal/proxy/task_index.go:153]: how to handle metric_type here?

**Pull request merged**